targets/TARGET_NORDIC/TARGET_NRF5/us_ticker.c

```diff
@@ -516,7 +542,7 @@ static void register_next_tick() {
     // the RTC1 keeps running.
     // This code is very short 20-38 cycles in the worst case, it shouldn't
     // disturb softdevice.
-    __disable_irq();
+    core_util_critical_section_enter();
```

Comment:
This might be capture here in the comment why disable irq are used here. I was not aware of this for instance, and thus asked why not critical section used here instead